flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisStreamsSink.java

```diff
@@ -89,13 +92,23 @@
                 maxBatchSizeInBytes,
                 maxTimeInBufferMS,
                 maxRecordSizeInBytes);
-        this.streamName =
-                Preconditions.checkNotNull(
-                        streamName,
-                        "The stream name must not be null when initializing the KDS Sink.");
-        Preconditions.checkArgument(
-                !this.streamName.isEmpty(),
-                "The stream name must be set when initializing the KDS Sink.");
+
+        // Ensure that either streamName or streamArn is set. If both are set, streamArn takes
+        // precedence.
+        if (streamArn != null) {
```

Comment:
Do you think this can be reusable by firehose, maybe we should move to aws-util

Reply:
mmm... We could move it once we re-use the logic.